### PR TITLE
Fix: Attempted to read or write protected memory. This is often an indication that other memory is corrupt

### PR DESCRIPTION
**Pull request by Kyrmasch**

### CODE REVIEW

File: NKalkan/KalkanValidationType.cs [hunk 1]
```diff
@@ -4,5 +4,6 @@ public enum KalkanValidationType
 {
     Nothing = 0x00000401,
     Crl = 0x00000402,
-    Ocsp = 0x0000040,
+    // FIx: Attempted to read or write protected memory. This is often an indication that other memory is corrupt
```
Comment (kant2002) [inline NKalkan/KalkanValidationType.cs]: This comment is not needed. I just pass incorrect value here.

File: NKalkan/KalkanApi.cs [hunk 2]
```diff
@@ -181,6 +181,10 @@ public void ValidateCertificate(string certificate, KalkanValidationType validat
             ThrowIfError(errorCode);
         }
 
+        // Fix: BUFFER_TOO_SMALL
```
Comment (kant2002) [inline NKalkan/KalkanApi.cs:184]: So you are saying that Kaplan report incorrect value? Did you have Latin text there or is there some Cyrillic characters?

File: NKalkan/KalkanApi.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ public sealed class KalkanApi
 
     static unsafe KalkanApi()
     {
-#if NETCOREAPP3_0_OR_GREATER
+        #if NETCOREAPP3_0_OR_GREATER
```
Comment (kant2002) [inline NKalkan/KalkanApi.cs]: Please undo these changes.

File: NKalkan/KalkanApi.cs (hunk 2 above)
Comment (Kyrmasch) [reply NKalkan/KalkanApi.cs:184]: If you do not increase the value, the message BUFFER_TOO_SMALL is constantly displayed. I looked at your changes (+20) - but it doesn't work.

File: NKalkan/KalkanApi.cs (hunk 3 above)
Comment (Kyrmasch) [reply NKalkan/KalkanApi.cs:21]: Ok

File: NKalkan/KalkanApi.cs (hunk 2 above)
Comment (Kyrmasch) [reply NKalkan/KalkanApi.cs:184]: `outputInformation:` OCSP: check certificate status: good
	certificateSerialNumber=50B4B1547BEB730C70CEBB817F3FE802E384704B
	This Update: 18.01.2024 14:57:09 ALMT
Verify chain and certificates: - OK.

`ospResponce`: OCSP Response Data:
    OCSP Response Status: successful (0x0)
    Response Type: Basic OCSP Response
    Version: 1 (0x0)
    Responder Id: CN = OCSP RESPONDER, C = KZ, O = \D2\B0\D0\9B\D0\A2\D0\A2\D�g���

Comment (kant2002) [reply NKalkan/KalkanApi.cs:184]: Let's duplicate on 2. I think that would be slightly more safe in case they count number of Cyrillic characters, but forgte take into account encoding.

Comment (Kyrmasch) [reply NKalkan/KalkanApi.cs:184]: I checked the full length of `ospResponseBuilder` - its value is 8006. Looked at the test cases in SDK (Test_const.cpp) and they have a length value of 8192 (LENGHT_STAND global variable). Don't know how to calculate correctly. With this length, `ospResponseBuilder` does not cause errors, it simply cuts out, but `outputInformationBuilder` is critical, if it is not enough, then the `BUFFER_TOO_SMALL` error occurs.

image

**Comment by kant2002** [comment]
Рахмет!

**Comment by kant2002** [comment]
Release 0.5.0
